### PR TITLE
migrate utxos to hooks

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/hooks/index.ts [hunk 1]
```diff
@@ -68,6 +68,42 @@ export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
...
 
+export const useUtxos = (
+  wallet: YoroiWallet,
+  options?: UseQueryOptions<Array<RawUtxo>, Error, Array<RawUtxo>, [string, 'utxos']>,
+) => {
+  const {refetch, ...query} = useQuery({
+    ...options,
+    refetchInterval: 20000,
+    queryKey: [wallet.id, 'utxos'],
+    queryFn: () => wallet.fetchUTXOs(),
```
Comment (stackchain) [inline src/hooks/index.ts:79]: - react-query has that issue with `refetchInterval` shall we ignore errors here? Or let the boundary kicks it?

File: src/hooks/index.ts [hunk 2]
```diff
@@ -68,6 +68,42 @@ export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
...
+  })
+
+  const netInfo = useNetInfo()
+  const onlineRef = React.useRef()
```
Comment (stackchain) [inline src/hooks/index.ts]: is this in use?

File: src/hooks/index.ts [hunk 3]
```diff
@@ -797,36 +836,10 @@ export const useExchangeRate = ({
...
-
-      return wallet.fetchUTXOs().then((utxos) => Utxos.toAmounts(utxos, primaryTokenId))
-    },
-  })
+  const {utxos} = useUtxos(wallet, {suspense: true})
+  if (utxos == null) throw new Error('invalid state')
 
-  const netInfo = useNetInfo()
-  const onlineRef = React.useRef()
+  const primaryTokenId = getDefaultAssetByNetworkId(wallet.networkId).identifier
```
Comment (stackchain) [inline src/hooks/index.ts:838]: make sense to useTokenInfo ?

**Review by stackchain — COMMENTED** [review]
⬆️

File: src/hooks/index.ts (hunk 1 above)
Comment (wolverineks) [reply src/hooks/index.ts:79]: what issue?

File: src/hooks/index.ts (hunk 2 above)
Comment (wolverineks) [reply src/hooks/index.ts]: yes

File: src/hooks/index.ts (hunk 3 above)
Comment (wolverineks) [reply src/hooks/index.ts:838]: i dont think soa

File: src/hooks/index.ts (hunk 1 above)
Comment (stackchain) [reply src/hooks/index.ts:79]: retries + onError https://github.com/TanStack/query/issues/859 , will affect this impl ? https://emurgo.slack.com/archives/C03GBAG4YAE/p1664393567296799

File: src/hooks/index.ts [hunk 4]
```diff
@@ -68,6 +68,42 @@ export const useWallet = (wallet: YoroiWallet, event: WalletEvent['type']) => {
...
+    queryKey: [wallet.id, 'utxos'],
+    queryFn: () => wallet.fetchUTXOs(),
+  })
+
+  const netInfo = useNetInfo()
+  const onlineRef = React.useRef()
+
+  React.useEffect(() => {
+    const isOnline = netInfo.type !== 'none' && netInfo.type !== 'unknown'
+    if (onlineRef.current !== isOnline) {
```
Comment (stackchain) [inline src/hooks/index.ts]: where `.current` is set?

**Review by banklesss — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

**Pull request merged**